python/measurement/contextual_sidebar_metrics.py

```diff
@@ -0,0 +1,45 @@
+import gzip
+import ijson
+import progressbar
+from collections import defaultdict
+from lib.helpers import dig
+from lib import services
+from data_extraction.export_data import jenkins_compatible_progress_bar
+from data import items_from_content_file
+
+def contextual_sidebar_metrics():
+    progress_bar = jenkins_compatible_progress_bar()
+    content_items = items_from_content_file()
+
+    nav_type_count = defaultdict(int)
+
+    for content_item in progress_bar(content_items):
+        nav_type_count[navigation_type(content_item)] += 1
+
+    for nav_type, count in nav_type_count.items():
+        services.statsd.gauge('related_navigation.' + nav_type, count)
+        print("{}: {}".format(nav_type, count))
+
+def navigation_type(content_item):
+    mainstream_tags = dig(content_item, 'links', 'mainstream_browse_pages') or []
+    curated_items = dig(content_item, 'links', 'ordered_related_items') or []
+
+    step_by_step = dig(content_item, 'links', 'part_of_step_navs') or []
+    if len(step_by_step) != 1:
+        step_by_step = None
+
+    live_taxons = any(
+        taxon['phase'] == 'live'
+        for taxon in dig(content_item, 'links', 'taxons') or []
+    )
+
+    if step_by_step:
+        return 'step_by_step'
+    elif mainstream_tags:
+        return 'mainstream'
+    elif curated_items:
+        return 'curated'
+    elif live_taxons:
+        return 'taxon'
+    else:
+        return 'default'
```

Comment:
Rather than defining the variables, and then having the `if`, perhaps consider merging them? e.g.

```python
...

if dig(content_item, 'links', 'mainstream_browse_pages'):
    return 'mainstream'

...
```

Reply:
Hmm, i think this would make the code more unreadable. I think this would be a good idea if retrieving the tags were as simple as it is for `mainstream_browse_pages` and `ordered_related_items` ?

Reply:
Ok, sure :+1: